tests/translate/translate_d2a2c_vect.py

```diff
@@ -18,14 +69,48 @@ def __init__(self, grid):
         }
         self.in_vars["parameters"] = ["dord4"]
         self.out_vars = {
-            "uc": grid.x3d_domain_dict(),
-            "vc": grid.y3d_domain_dict(),
+            # "uc": grid.x3d_domain_dict(),
+            # "vc": grid.y3d_domain_dict(),
             "ua": {},
             "va": {},
-            "utc": {},
-            "vtc": {},
+            # "utc": {},
+            # "vtc": {},
         }
         # TODO: This seems to be needed primarily for the edge_interpolate_4
         # methods, can we rejigger the order of operations to make it match to
         # more precision?
         self.max_error = 2e-10
+
+    def compute(self, inputs):
+        if spec.namelist.npx != spec.namelist.npy:
+            raise NotImplementedError("D2A2C assumes a square grid")
+        if spec.namelist.npx <= 13 and spec.namelist.layout[0] > 1:
+            D2A2C_AVG_OFFSET = -1
+        else:
+            D2A2C_AVG_OFFSET = 3
+
```

Comment:
I'm so glad this works. Our tests currently don't cover everything here, so would you mind manually testing c12_54ranks and c48_54ranks one more time? I'd be happy to do this for you later if you'd like.

Reply:
Good idea. I'm running those now before merging the PR.